src/main/scala/org/constellation/consensus/EdgeProcessor.scala

```diff
@@ -248,22 +259,80 @@ object EdgeProcessor extends StrictLogging {
     result.sequence
   }
 
-  // Temporary for testing join/leave logic.
+  def resolveTransactions(
+    hashes: Seq[String],
+    priorityPeer: Id
+  )(implicit dao: DAO): Future[Seq[TransactionCacheData]] = {
 
-  def handleSignatureRequest(sr: SignatureRequest)(implicit dao: DAO): SignatureResponse = {
-    //if (sr.facilitators.contains(dao.id)) {
-    // val replyTo = sr.checkpointBlock.witnessIds.head
-    val updated = if (sr.checkpointBlock.simpleValidation()) {
-      Some(hashSign(sr.checkpointBlock.baseHash, dao.keyPair))
-    } else {
-      None
+    implicit val exec: ExecutionContextExecutor = dao.signatureExecutionContext
+
+    def lookupTransaction(hash: String, client: APIClient): Future[Option[TransactionCacheData]] = {
+      async {
+        await(
+          client.getNonBlocking[Option[TransactionCacheData]](
+            s"transaction/$hash",
+            timeout = 4.seconds
+          )
+        )
+      }
+    }
+
+    def resolveTransaction(hash: String,
+                           peers: Seq[APIClient]): Future[Option[TransactionCacheData]] = {
+
+      var remainingPeers = peers.tail
+
+      lookupTransaction(hash, peers.head).transformWith {
+        case Success(Some(transactionCacheData)) => Future.successful(Some(transactionCacheData))
+        case _ =>
```

Comment:
since in the successful case you're keeping the value the same, I think you really want .recoverWith instead of .transformWith.

Also -- a cool pattern matching trick if you were going to do transformWith is labeling the match.

```case s @ Success(Some(transactionCacheData)) => s```

Reply:
I need a Future type though returned not a Try -- but I'm changing this to recoverWith cause its cleaner.